logstash-core/src/main/java/org/logstash/config/ir/graph/Edge.java

```diff
@@ -22,15 +39,29 @@ public void setGraph(Graph graph) {
         }
     }
 
+    @Override
+    public final int hashCode() {
+        return 37 * from.hashCode() + to.hashCode();
```

Comment:
@andrewvc I tend to not use that method (doesn't matter all that much here), it's really slow compared to just doing you own thing because of the varargs triggering array creation on every call + it's meh to read, now I can't just click on `hashCode()` for `from` and `to` in the IDE and have to go another step if I wanted to read through this and make sure everything is on the up and up again :D